pkg/reconciler/managed/reconciler.go

```diff
@@ -693,7 +694,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	// not realize that the controller is still trying to reconcile
 	// (and modify or delete) the resource since they forgot to enable the
 	// feature flag.
-	if !r.managementPoliciesEnabled && (managed.GetManagementPolicy() == xpv1.ManagementObserveOnly || managed.GetManagementPolicy() == xpv1.ManagementOrphanOnDelete) {
+	if !r.managementPoliciesEnabled && (managed.GetManagementPolicy() != nil && !ContainsOnlyManagementAction(managed.GetManagementPolicy(), xpv1.ManagementActionAll)) {
```

Comment:
Starting to worry about readability in general. I am wondering if we could improve it by wrapping the logic in function(s) and also defining some internal (whitelisted) combinations as states. 

I don't have a great example of what it could look like yet :)